contributing/deprecation_policy.md

```diff
@@ -38,51 +128,70 @@ A three-month deprecation is similar with more time for clients to adjust code:
+Body:
+YYY will eventually be deprecated. Please see <link to migration guide> for more details.
 
-This macro requires an explanation string which should be similar to: ”This API will be removed on D/M/YYYY. Please use XYZ instead. See issue #n for more details.”
+The timeline for this deprecation is as follows:
 
-For any code that should be eventually deleted, add the following comment nearby to make it easier to find and remove the code: // DEPRECATION(github.com/material-components/material-components-ios/issues/n)
+- <Deprecation date>: The API will be formally marked as deprecated. You will start receiving
+  build warnings if you are still using the deprecated API.
+- <Deletion date>: The API will be deleted.
+```
 
-Once you submit the breaking PR, don’t immediately pile on subsequent PRs that would conflict with an emergency rollback.
+### Step 7: After the deprecation date, deprecate the API
 
-## Versioning
+Send a pull request annotating the API as deprecated. Use `__deprecated_msg("")` and provide a
+concise message with a recommendation for an alternative API, if any is available.
 
-Adding the deprecation *warning* (no breaking change) is a minor version bump.
+### Step 8: After the deletion date, re-evaluate usage of the API
 
-Actually changing the code (breaking change) is a major version bump.
+Once the deletion date has passed and at least one release has been cut since the API was marked as
+deprecated, you can now consider deleting the API. But first: you must evaluate whether the API is
+being used by any internal clients.
 
-Changes that don't break anyone (or change behavior surprisingly) don't need a deprecation.
+If the API is still being used by clients, you have a few options:
 
-## Checklist
+- Reach out to the client teams directly, reminding them of the deprecation.
+- Help the client teams migrate.
 
-This is an idealized summary of how to replace old code with new code.
+Proceed to step 9 once you've confirmed that there are zero internal usages of the API.
 
-1. Create an alternative.
-    1. Discuss the reasoning via GitHub issue.
-    1. Discuss the change via GitHub pull request.
-1. Publicize and do the deprecation.
-    1. Document the old code so new users don’t start using it.
-    1. Create a breaking change schedule. 
-    1. Mark the old code as deprecated.
-1. Clean up
-    1. Delete the old code.
-    1. Close the issue.
-    1. Release
+### Step 9: Delete the API and migration guide
+
+Once you have confirmed that there is no internal usage of the API, you can safely delete it and the migration guide.
+
+Send a pull request deleting the API, marking your "Delete YYY" issue as closed.
+
+## Addendum
+
+### Versioning
+
+Adding the deprecation *warning* (no breaking change) is a minor version bump.
+
+Actually changing the code (breaking change) is a major version bump.
+
+Changes that don't break anyone (or change behavior surprisingly) don't need a deprecation.
```

Comment:
Should we try to group "breaking changes" of different APIs together by designating preferred deletion dates and possibly preferred versions for breaking changes? It might make it easier for our client base to plan ahead for these dates/versions.